mlrun/projects/project.py

```diff
@@ -3192,7 +3212,9 @@ def _init_function_from_obj(
         func.metadata.project = project.metadata.name
     if project.spec.tag:
         func.metadata.tag = project.spec.tag
-    return name or func.metadata.name, func
+    if name:
+        func.metadata.name = name
+    return func.metadata.name, func
```

Comment:
Note that previously, when setting a function from function object, it would keep the old function name and use the "name" param as key in the project context. IMO it's a bug but this fix breaks BC so we might prefer not to do it.